### PR TITLE
chat: properly parse email addresses with +

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/libkb/util.go [hunk 1]
```diff
@@ -1034,8 +1034,8 @@ var kbfsPathInnerRegExp = func() *regexp.Regexp {
 	// e.g. alice@twitter
 	const regularAssertion = `[-_a-zA-Z0-9.+]+@[a-zA-Z.]+`
 	// e.g. [bob@keybase.io]@email
-	const atContainingAssertion = `\[[-_a-zA-Z0-9.]+@[-_a-zA-Z0-9.]+\]@[a-zA-Z.]+`
-	const socialAssertion = `(?:` + regularAssertion + `)|(?:` + atContainingAssertion + `)`
+	const emailAssertion = `\[[-_+a-zA-Z0-9.]+@[-_a-zA-Z0-9.]+\]@[a-zA-Z.]+`
```
Comment (strib) [inline go/libkb/util.go:1037]: I could be mistaken, but don't we need a `\` before the `+`?  Otherwise this means one or more underscores?

Comment (songgao) [reply go/libkb/util.go:1037]: I think the fact that it's inside `[]` removes the need for `/`. I tested in chat to make sure it worked, but will also add some tests for this.

Comment (strib) [reply go/libkb/util.go:1037]: Cool thanks.  That's confusing.  Stupid regexes.

Comment (songgao) [reply go/libkb/util.go:1037]: 😆

**Pull request merged**